api/CcsSso.Core.Service/External/OrganisationSiteService.cs

```diff
@@ -44,6 +44,21 @@ public async Task<int> CreateSiteAsync(string ciiOrganisationId, OrganisationSit
     {
       Validate(organisationSiteInfo);
 
+      //Validate Sitename duplication in same org
+      var old_sites = _dataContext.ContactPoint
+        .Include(cp => cp.ContactPointReason)
+        .Include(cp => cp.ContactDetail).ThenInclude(cd => cd.PhysicalAddress)
```

Comment:
Looks like similar code used elsewhere. Can you make it as a private function so you no need to repeat it.